go-controller/pkg/ovn/controller/apbroute/node_controller.go

```diff
@@ -369,10 +361,18 @@ func (c *ExternalGatewayNodeController) syncNamespace(namespace *v1.Namespace) e
 		// it's not a namespace being cached already and it is not a target for policies, nothing to do
 		return nil
 	}
+
+	defer c.mgr.unlockNamespaceInfoCache(namespace.Name)
+	if found && cacheInfo.markForDelete {
+		// namespace exists and has been marked for deletion, this means there should be an event to complete deleting the namespace.
+		// wait for the namespace to be deleted before recreating it in the cache.
+		klog.Infof("Attempting to add namespace %s when it has been marked for deletion. Waiting for deletion event to be processed")
```

Comment:
The error that you are returning should be logged somewhere so I don't think we need this info log

Reply:
deleted in both master and node controllers